src/ports/web/api/events/export.rs

```diff
@@ -0,0 +1,176 @@
+use super::*;
+
+#[get("/export/events.csv?<query..>")]
+pub fn csv_export(
+    connections: sqlite::Connections,
+    search_engine: tantivy::SearchEngine,
+    auth: Auth,
+    query: usecases::EventQuery,
+) -> result::Result<Content<String>, AppError> {
+    let events = query_events(connections, search_engine, auth, query)?;
+
+    let records: Vec<_> = events.map(adapters::csv::EventRecord::from).collect();
+
+    let buff: Vec<u8> = vec![];
+    let mut wtr = csv::Writer::from_writer(buff);
+
+    for r in records {
+        wtr.serialize(r)?;
+    }
+    wtr.flush()?;
```

Comment:
Is flushing really necessary here? I suppose into_inner() should do this implicitly. Otherwise, please add a comment why.